addons/project/models/project_task.py

```diff
@@ -1685,6 +1691,28 @@ def action_recurring_tasks(self):
             'domain': [('recurrence_id', 'in', self.recurrence_id.ids)],
         }
 
+    def action_project_sharing_recurring_tasks(self):
+        self.ensure_one()
+        recurrent_tasks = self.env['project.task'].search([('recurrence_id', 'in', self.recurrence_id.ids)])
+        # If all the recurrent tasks are in the same project, open the list view in sharing mode.
+        if recurrent_tasks.project_id == self.project_id:
+            action = self.env['ir.actions.act_window']._for_xml_id('project.project_sharing_project_task_recurring_tasks_action')
+            action.update({
+                'context': {'default_project_id': self.project_id.id},
+                'domain': [
+                    ('project_id', '=', self.project_id.id),
+                    ('display_in_project', '=', True),
+                    ('recurrence_id', 'in', self.recurrence_id.ids)
+                ]
+            })
+            return action
+        # If at least one recurrent task belong to another project, open the portal page
+        return {
+            'name': 'Portal recurrent Tasks',
```

Comment:
It is used in the sub tasks one (maybe somewhere else, but i didn't find example). Since the two feature were similar, i based most of that part of the code on the way the sub tasks was handled.